whois-nrtm4/src/main/java/net/ripe/db/nrtm4/generator/SnapshotFileGenerator.java

```diff
@@ -91,10 +96,16 @@ public void createSnapshot()  {
 
         final Map<CIString, byte[]> sourceToOutputBytes = writeToGzipStream(snapshotState, sourceToNewVersion);
 
-        saveToDatabase(sourceToNewVersion, sourceToOutputBytes);
-        LOGGER.info("Snapshot generation complete {}", stopwatch);
+        try {
+            nrtmSourceContext.setCurrentSourceToWhoisMaster();
+
+            saveToDatabase(sourceToNewVersion, sourceToOutputBytes);
+            LOGGER.info("Snapshot generation complete {}", stopwatch);
 
-        cleanUpOldFiles();
+            cleanUpOldFiles();
+        } finally {
+            nrtmSourceContext.setCurrentSourceToWhoisSlave();
```

Comment:
In my opinion, we should not use sourceaware cotext for DB write, we should always specify write only datasource instead of changing it to master and later changing to slave as you ca  never use slave source to write to DB. May be there are other settings which made yo use sourceaware?

Reply:
sourceAware is doing on this way
Just copy paste the behaviour. Check singleUpdates class